test/lhash_test.c

```diff
@@ -555,16 +557,11 @@ static void do_mt_hash_work(void)
     HT_INIT_KEY(&key);
 
     for (iter = 0; iter < TEST_THREAD_ITERATIONS; iter++) {
-        if (!RAND_bytes((unsigned char *)&index, sizeof(unsigned int))) {
-            worker_exits[num] = "Failed to get random index";
+        if (!TEST_true(CRYPTO_THREAD_write_lock(testrand_lock)))
             return;
-        }
-        index %= TEST_MT_POOL_SZ;
-        if (!RAND_bytes((unsigned char *)&behavior, sizeof(char))) {
-            worker_exits[num] = "Failed to get random behavior";
-            return;
-        }
-        behavior &= BEHAVIOR_MASK;
+        index = test_random() % TEST_MT_POOL_SZ;
+        behavior = (char)(test_random() & BEHAVIOR_MASK);
```

Comment:
Suggest using `%` here to avoid confusing breaks when the macro is changed.

Reply:
Done

Reply:
Please re-approve